go-controller/pkg/libovsdb/util/mac_binding.go

```diff
@@ -48,3 +50,37 @@ func CreateMACBinding(sbClient libovsdbclient.Client, logicalPort, datapathName
 
 	return nil
 }
+
+// DeleteSbdbMacBindingsWithIPs deletes mac bindings from the SBDB on any of the
+// provided IPs. This method should only be used to remove legacy entries, as
+// static mac bindings should be created in NBDB instead.
+func DeleteSbdbMacBindingsWithIPs(sbClient libovsdbclient.Client, ips ...string) error {
+	mb := &sbdb.MACBinding{}
+	conditions := make([]model.Condition, len(ips))
+	for i := range ips {
+		conditions = append(conditions,
+			model.Condition{
+				Field:    &mb.IP,
+				Function: ovsdb.ConditionEqual,
+				Value:    ips[i],
+			},
+		)
+	}
+
+	// Delete using the client native API instead of model client to avoid
+	// having to monitor & cache the SBDB mac binding table.
+	// This will encode the conditions as provided and send them to the server
+	// if there is no cache. The operation is idempotent and there is no error
```

Comment:
would have been helpful to include this explanation in the commit message for people that dont know what a native api is

Reply:
wait a second...dont we block the creation of the sbClient on the monitor itself. In other words, we cant have the client unless monitor worked correctly and has already populated the cache. I thought we do this intentionally to avoid a situation where we execute client commands with a cache that is not populated. If that is true, then how does this native strategy make any difference? I would think it would only matter if a client was created without monitoring.

Reply:
Not sure I understand the concern. Perhaps I have not been clear enough with the comment.

In this PR, we no longer monitor the SBDB MAC Binding table, so we won't have it in the cache and thus we can use model client API as that relies on the cache: it looks up the model and if it doesn't find any, it won't send the delete towards the server.

As opposed to the native client API, which will encode and send the delete ops to the server as they have been provided to it regardless of whether the SBDB MAC Binding table is being monitored or not.

Reply:
yeah it made more sense as I got to the other commits. I think the initial commit message is what was confusing.